internal/cmd/controller/gitops/reconciler/status_controller.go

```diff
@@ -165,23 +190,80 @@ func setStatus(list *v1alpha1.BundleDeploymentList, gitrepo *v1alpha1.GitRepo) e
 	return nil
 }
 
+// setReadyStatusFromBundle fetches all bundles from a given gitrepo, checks the ready status conditions
+// from the bundles and applies one on the gitrepo if it isn't ready. The purpose is to make
+// rendering issues visible in the gitrepo status. Those issues need to be made explicitly visible
+// since the other statuses are calculated from bundle deployments, which do not exist when
+// rendering manifests fail. Should an issue be on the bundle, it will be copied to the gitrepo.
+func (r StatusReconciler) setReadyStatusFromBundle(ctx context.Context, gitrepo *fleet.GitRepo) error {
+	bList := &fleet.BundleList{}
+	err := r.List(ctx, bList, client.MatchingLabels{
+		fleet.RepoLabel: gitrepo.Name,
+	}, client.InNamespace(gitrepo.Namespace))
+	if err != nil {
+		return err
+	}
+
+	found := false
+	// Find a ready status condition in a bundle which is not ready.
+	var condition genericcondition.GenericCondition
+bundles:
+	for _, bundle := range bList.Items {
+		if bundle.Status.Conditions == nil {
+			continue
+		}
+
+		for _, c := range bundle.Status.Conditions {
+			if c.Type == string(fleet.Ready) && c.Status == v1.ConditionFalse {
+				condition = c
+				found = true
+				break bundles
```

Comment:
It looks like this will only propagate the _first_ bundle rendering error to the gitrepo status, if multiple such errors exist over multiple bundles.
Perhaps something to clarify in documentation.

Reply:
Yes, that's true. That is something that we might want to change at some point, but I'd need to think more about it, since there's [another possible optimization](https://github.com/rancher/fleet/pull/3114#discussion_r1878117893). In the meantime, I'm thinking of adding some documentation to our docs, too, since the errors we start showing in Bundle and GitRepo won't be found in Cluster or ClusterGroup.